### ISSUE TITLE
field with no_log still shown in -vvvv output

### ISSUE THREAD

**Issue by jschwanz** [ansible/ansible#6581]
##### Issue Type:

Bug Report
##### Ansible Version:

Ansible 1.5.2
##### Environment:

RHEL 6
##### Summary:

In developing a module I've added no_log=True to some fields in the argument_spec. When running a play/playbook with the module using -vvvv I am seeing my "private" values in the output.
##### Steps To Reproduce:

Here is a basic version of my module.

``` python
class MyClass(object):
    def __init__(self, module):
        self.module = module
        self.pub1 = module.params['pub1']
        self.pub2 = module.params['pub2']
        self.priv = module.params['priv']

    def dostuff(self):
        return self.module.run_command(something)


def main():
    module = AnsibleModule(
        argument_spec = dict(
            pub1=dict(),
            pub2=dict(),
            priv=dict(no_log=True),
        )
    )

    mine = MyClass(module)
    result = mine.dostuff()
    module.exit_json(**result)

from ansible.module_utils.basic import *
main()
```
##### Expected Results:

```
<hostname> REMOTE_MODULE mymodule pub1=value1 pub2=value2 
```
##### Actual Results:

```
<hostname> REMOTE_MODULE mymodule pub1=value1 pub2=value2 priv=privvalue
```

**Comment by mpdehaan** [comment]
no_log=True on the argument spec is intended for syslog, so this is an RFE to hide it from ansible CLI output itself, which I approve of.

It is suggested that the module itself return a list of parameters that should not be logged as "no_log=True", that ansible never records this, but uses it to filter any output to the callbacks.

Must be done in a way that doesn't break any existing callbacks.

**Comment by mpdehaan** [comment]
Implemented on 1.8 devel here:

https://groups.google.com/forum/#!topic/ansible-project/b-nZj_RpxFY
